Corrected two return types in Area2D

MattUV opened the pull request:
Corrected the return type of overlaps_body() and overlaps_area() in the Area2D class documentation: they are now booleans.

This is my first attempt with github, I hope I did it correctly. (I can't access the collaborative pad, due to 504 error...)

MattUV committed:
Corrected two return types in Area2D

Corrected the return type of overlaps_body() and overlaps_area() in Area2D class
They are now booleans

akien-mga commented:
Nice catch! Sadly it won't work to change it this way, the return types in the XML are inferred directly from the source code. So you would have to fix the binding of those two functions in the C++, i.e.:
- By removing the ``:PhysicsBody2D`` and ``:Area2D``` parts here (``bool`` is inferred automatically): https://github.com/godotengine/godot/blob/a895e2e372c9df2a9972f38134ae0ea9c8bf45a8/scene/2d/area_2d.cpp#L641-L642
- Same thing for 3D: https://github.com/godotengine/godot/blob/e07266f6c1177d963f015fa108af40a315305f89/scene/3d/area.cpp#L631-L632

MattUV committed:
Corrected two bind_methods

Based on this discussion : https://github.com/godotengine/godot/pull/4635#issuecomment-218753369

MattUV committed:
Corrected to binding_methods

Based on this discussion : https://github.com/godotengine/godot/pull/4635#issuecomment-218753369

MattUV commented:
Ok, thank you. I edited these two files. However, I didn't find the way to remove my modification request for classes.xml ...

akien-mga commented:
It's fine, as what you modified manually is now what the `doctool` would have generated. I'll just squash the three commits together on merge to reduce the verbosity :)
